src/FakeItEasy/OutputWriterExtensions.cs

```diff
@@ -49,5 +51,70 @@ public static IOutputWriter Write(this IOutputWriter writer, object value)
             writer.Write(value.ToString());
             return writer;
         }
+
+        /// <summary>
+        /// Formats the specified argument values as strings and writes
+        /// them to the output, separated by the specified separator.
+        /// </summary>
+        /// <param name="writer">The writer to write to.</param>
+        /// <param name="values">The values to write to the writer.</param>
+        /// <param name="skipFormatting">Specify that argument values should not be formatted.</param>
+        /// <returns>The writer.</returns>
+        internal static IOutputWriter WriteArgumentValues(this IOutputWriter writer, IEnumerable values, bool skipFormatting = false)
+        {
+            Guard.AgainstNull(writer, nameof(writer));
+            Guard.AgainstNull(values, nameof(values));
+
+            var list = values.AsList();
+            if (list.Count <= 5)
+            {
+                writer.WriteArgumentValuesImpl(list, skipFormatting);
+            }
+            else
+            {
+                writer.WriteArgumentValuesImpl(list.Take(2), skipFormatting);
+                int remainingCount = list.Count - 4;
+                writer.Write($", … ({remainingCount} more elements) …, ");
+                writer.WriteArgumentValuesImpl(list.Skip(list.Count - 2), skipFormatting);
+            }
+
+            return writer;
+        }
+
+        /// <summary>
+        /// Formats the specified argument values as strings and writes
+        /// them to the output, separated by the specified separator.
+        /// </summary>
+        /// <param name="writer">The writer to write to.</param>
+        /// <param name="values">The values to write to the writer.</param>
+        /// <param name="skipFormatting">Specify that argument values should not be formatted.</param>
+        /// <returns>The writer.</returns>
+        private static IOutputWriter WriteArgumentValuesImpl(this IOutputWriter writer, IEnumerable values, bool skipFormatting = false)
+        {
+            Guard.AgainstNull(writer, nameof(writer));
+            Guard.AgainstNull(values, nameof(values));

```

Comment:
Are the guards necessary for a private method (I often wonder the same thing about the internal methods as well).

Reply:
No, probably not. I'll remove them.
The way I see it, the guards are necessary for public methods to ensure the caller is using the method correctly. Internally, we're supposed to know how to use it, so if we pass a null, it's our own problem, and there's no point in throwing a specific exception. I haven't always applied this rule, though...